main/src/cgeo/geocaching/utils/MapMarkerUtils.java

```diff
@@ -697,4 +705,41 @@ public static void resetLists() {
         return result;
     }
 
+    public static Drawable getDTRatingMarker(final Resources res, final float difficulty, final float terrain) {
+        final int hashcode = new HashCodeBuilder().append(difficulty).append(terrain).toHashCode();
+
+        synchronized (overlaysCache) {
+            CacheMarker marker = overlaysCache.get(hashcode);
+            if (marker == null) {
+                marker = new CacheMarker(hashcode, createDTRatingMarker(res, difficulty, terrain));
+                overlaysCache.put(hashcode, marker);
+            }
+            return marker.getDrawable();
+        }
+    }
+
+    private static LayerDrawable createDTRatingMarker(final Resources res, final float difficulty, final float terrain) {
+        final Drawable background = DrawableCompat.wrap(ResourcesCompat.getDrawable(res, R.drawable.marker_rating_bg, null));
+        final InsetsBuilder insetsBuilder = new InsetsBuilder(res, background.getIntrinsicWidth(), background.getIntrinsicHeight());
+        insetsBuilder.withInset(new InsetBuilder(background));
+
+        final String packageName = CgeoApplication.getInstance().getPackageName();
+
+        Map<String, Integer> dt = new HashMap<>();
+        dt.put("d", (int) difficulty * 10);
+        dt.put("t", (int) terrain * 10);
+
+        for (Map.Entry<String, Integer> p: dt.entrySet()) {
+            if (p.getValue() >= 10) {
```

Comment:
I assume that the value is < 10 for caches without d/t rating (e.g. labs) correct? What happens in this case? Is an empty drawable created?

Reply:
The drawable is created with 2 empty sections as I didn't want to create an empty "0" drawable.

Reply:
Had to add them now though to be eliminate the map ...